src/presign/participant.rs

```diff
@@ -389,7 +549,18 @@ impl PresignParticipant {
             ));
         }
 
-        Ok(ret_messages)
+        // Additionally, handle any round 3 messages which may have been received too early
+        let mut presign_record_option = None;
+        let retrieved_messages = self.fetch_messages(
+            MessageType::Presign(PresignMessageType::RoundThree),
+            message.id(),
+        )?;
+        for msg in retrieved_messages.iter() {
+            let (pr_option, _) = self.handle_round_three_msg(rng, msg, main_storage)?;
+            presign_record_option = pr_option;
```

Comment:
question: Same as above -- won't this get overwritten if you've gotten more than one round three message? It looks like `handle_round_three_msg` can actually return `Some`, sometimes.

My guess is that there's an unstated assumption here that order matters and only the "last" round 3 message will ever return `Some`, so it's fine to overwrite all previous outputs. Is that true?

Reply:
Yep, your intuition is correct. I think I'll still change this to do some defensive programming